git/client.go

```diff
@@ -48,13 +45,9 @@ type RepositoryWriter interface {
 	// SwitchBranch switches from the current branch of the repository to the
 	// provided branch. If the branch doesn't exist, it is created.
 	SwitchBranch(ctx context.Context, branch string) error
-	// WriteFile creates a new file at the given path, reads from the reader and
-	// writes the content to the file. If the file already exists, its overwritten.
-	WriteFile(path string, reader io.Reader) error
-	// Commit commits any changes made to the repository.
-	// signer is an optional argument which can be provided to sign the commit using
-	// OpenPGP.
-	Commit(info Commit, signer *openpgp.Entity) (string, error)
+	// Commit commits any changes made to the repository. commitOpts is an
+	// optional argument which can be provided to configure the commit.
+	Commit(info Commit, commitOpts *CommitOptions) (string, error)
```

Comment:
This is fine but I was thinking more in terms of functional options, for example the API of the patch helper in https://github.com/fluxcd/pkg/blob/10b923813d45de15d052264f7fa345ab21a6c1b2/runtime/conditions/patch.go#L125.
A good description of this pattern is https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis.
This way, you don't even have to pass anything, really optional.